src/movies.js

```diff
@@ -1,16 +1,103 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+
+function getAllDirectors(movieArray) {
+    let directors = movieArray.map(function(item){
+        return item.director;
+    })
+}
+    
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movieArray){
+    let filteredList = movieArray.filter(function(item){
+        return item.["director"] === "Steven Spielberg" && item.["genre"].includes("Drama")
+    })
+    return filteredList.length;
+}
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movieArray) {
+    if (movies.length == 0){
```

Comment:
remember to use triple equal instead of double equal when comparing. Best practices are important!